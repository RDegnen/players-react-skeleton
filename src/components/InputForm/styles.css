.formCol {
  padding-left: 30px; 
}