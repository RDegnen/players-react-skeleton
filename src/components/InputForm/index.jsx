import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Form, FormGroup, Label, Input, Col, Row } from 'reactstrap';

import componentProps from '../props';
import './styles.css';

const InputForm = ({ formGroups, handleInput, handleSubmit }) => (
  <Row>
    <Col m="6" className="formCol">
      <Form onSubmit={handleSubmit}>
        {formGroups.map(({
          field, type, id, label, renderGroup,
        }) => (
          <Fragment key={field}>
            {renderGroup ? (
              renderGroup()
            ) : (
              <FormGroup key={`${field}Group`} row>
                <Label key={`${field}Label`} for={field}>{label}</Label>
                <Input
                  key={`${field}Input`}
                  type={type}
                  name={field}
                  id={id}
                  placeholder={label}
                  onChange={handleInput}
                />
              </FormGroup>
            )}
          </Fragment>
        ))}
      </Form>
    </Col>
    <Col m="6" />
  </Row>
);

InputForm.propTypes = {
  formGroups: componentProps.formGroupShape.isRequired,
  handleInput: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default InputForm;
