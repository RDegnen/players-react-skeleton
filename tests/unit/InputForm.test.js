import React from 'react';
import { shallow } from 'enzyme';
import { FormGroup } from 'reactstrap';

import InputForm from '../../src/components/InputForm';

const defaultProps = {
  formGroups: [
    {
      field: 'email',
      type: 'email',
      id: 'email',
      label: 'Email',
    },
    {
      field: 'password',
      type: 'password',
      id: 'password',
      label: 'Password',
    },
    {
      renderGroup: () => (
        <FormGroup>
          Test
        </FormGroup>
      ),
    },
  ],
  handleInput: jest.fn(),
  handleSubmit: jest.fn(),
};

const getComponent = props => shallow(<InputForm {...defaultProps} {...props} />);

describe('<InputForm />', () => {
  it('should match snapshot', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should have the correct amount of FormGroups', () => {
    const component = getComponent();
    expect(component.find(FormGroup).length).toBe(3);
  });
});
