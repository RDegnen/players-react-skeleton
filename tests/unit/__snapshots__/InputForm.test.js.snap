// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputForm /> should match snapshot 1`] = `
<Row
  tag="div"
>
  <Col
    className="formCol"
    m="6"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Form
      onSubmit={[MockFunction]}
      tag="form"
    >
      <FormGroup
        key="emailGroup"
        row={true}
        tag="div"
      >
        <Label
          for="email"
          key="emailLabel"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Email
        </Label>
        <Input
          id="email"
          key="emailInput"
          name="email"
          onChange={[MockFunction]}
          placeholder="Email"
          type="email"
        />
      </FormGroup>
      <FormGroup
        key="passwordGroup"
        row={true}
        tag="div"
      >
        <Label
          for="password"
          key="passwordLabel"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Password
        </Label>
        <Input
          id="password"
          key="passwordInput"
          name="password"
          onChange={[MockFunction]}
          placeholder="Password"
          type="password"
        />
      </FormGroup>
      <FormGroup
        tag="div"
      >
        Test
      </FormGroup>
    </Form>
  </Col>
  <Col
    m="6"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  />
</Row>
`;
